### ISSUE TITLE
Crash while unzip this 7z file

### ISSUE THREAD

**Issue by xVanTuring** [miurahr/py7zr#181]
**Describe the bug**
Crash while unzip this file 

**Reproduce**
just unzip it with "py7zr x ./rock_cliff_stylized_mossy.7z"
[rock_cliff_stylized_mossy.zip](https://github.com/miurahr/py7zr/files/4891448/rock_cliff_stylized_mossy.zip)
(it's 7z but github can upload zip only)

**Expected behavior**
Should be extracted...

**Environment**
 - OS: Ubuntu 20.04
 - Python Python 3.8
 - py7zr version: 0.8.0

**Crash Report**
```
Traceback (most recent call last):
  File "/home/xvan/.local/bin/py7zr", line 8, in <module>
    sys.exit(main())
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/__main__.py", line 26, in main
    return cli.Cli().run()
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/cli.py", line 92, in run
    return args.func(args)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/cli.py", line 281, in run_extract
    a.extractall(callback=cb)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/py7zr.py", line 629, in extractall
    self._extract(path=path, return_dict=False, callback=callback)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/py7zr.py", line 735, in _extract
    self.worker.extract(self.fp, parallel=(not self.password_protected and not self._filePassed))
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/py7zr.py", line 950, in extract
    self.extract_single(fp, self.files, self.src_start, src_end, q)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/py7zr.py", line 993, in extract_single
    crc32 = self.decompress(fp, f.folder, ofp, f.uncompressed, f.compressed, src_end)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/py7zr.py", line 1023, in decompress
    decompressor = folder.get_decompressor(compressed_size)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/archiveinfo.py", line 381, in get_decompressor
    self.decompressor = SevenZipDecompressor(self.coders, packsize, self.unpacksizes, self.crc, self.password)
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/compressor.py", line 365, in __init__
    self.methods_map = [SupportedMethods.is_native_coder(coder) for coder in coders]  # type: List[bool]
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/compressor.py", line 365, in <listcomp>
    self.methods_map = [SupportedMethods.is_native_coder(coder) for coder in coders]  # type: List[bool]
  File "/home/xvan/.local/lib/python3.8/site-packages/py7zr/compressor.py", line 573, in is_native_coder
    raise UnsupportedCompressionMethodError
py7zr.exceptions.UnsupportedCompressionMethodError

```

**Comment by xVanTuring** [comment]
Here is output from 7z 
The method is 'Copy'
```
Extracting archive: ./rock_cliff_stylized_mossy.7z
--
Path = ./rock_cliff_stylized_mossy.7z
Type = 7z
Physical Size = 4113734
Headers Size = 218
Method = Copy
Solid = +
Blocks = 1

Everything is Ok

Files: 2
Size:       4113516
Compressed: 4113734

```

**Comment by miurahr** [comment]
As message said, COPY is 'Unsupported Compression Method'
It should not show a trace.

**Comment by miurahr** [comment]
A file you post has a compression method b'' == NULL
I have never seen it.

**Comment by miurahr** [comment]
7z specification defines COPY method as 0x00

https://github.com/szcnick/p7zip/blob/dev/DOC/Methods.txt#L28-L31
```
List of defined IDs
-------------------
      
00 - Copy
```
so method == NULL is breaking a specification. It seems a C/C++ bug?

**Comment by miurahr** [comment]
in p7zip implementation, method id is retrieved as following logics
```
  UInt32 i;
  for (i = 0; i < numCoders; i++)
  {
    CCoderInfo &coder = folder.Coders[i];
    {
      Byte mainByte = ReadByte();
      if ((mainByte & 0xC0) != 0)
        ThrowUnsupported();
      unsigned idSize = (mainByte & 0xF);
      if (idSize > 8 || idSize > GetRem())
        ThrowUnsupported();
      const Byte *longID = GetPtr();
      UInt64 id = 0;
      for (unsigned j = 0; j < idSize; j++)
        id = ((id << 8) | longID[j]);
      SkipDataNoCheck(idSize);
      coder.MethodID = id;
```
A default value is  ` UInt64 id = 0;` so NULL method become 0x00.

**Comment by miurahr** [comment]
We can change py7zr logic from 
```
            c = {'method': file.read(methodsize)} 
```

to

```
            c = {'method': file.read(methodsize) if methodsize > 0 else 0} 
```

https://github.com/miurahr/py7zr/blob/master/py7zr/archiveinfo.py#L313

**Comment by miurahr** [comment]
@xVanTuring Could you make a small test data for regression test?

It may not able to create it on Linux. 
A sample is https://github.com/miurahr/py7zr/blob/master/tests/data/copy.7z

**Comment by xVanTuring** [comment]
@miurahr  Hey, so this weird zip package was generated by Substance Designer, so I can provide a default empty package from it, it should be small enough, but the content may not be changeable.
[Empty.zip](https://github.com/miurahr/py7zr/files/4894443/Empty.zip)
(again can't upload with 7z extension)

**Comment by xVanTuring** [comment]
After changing the line as you mentioned, the error still exists

**Comment by miurahr** [comment]
Thanks for data. I've pushed the fix as #182

I've changed a line like 
```
        if methodsize > 0
                c = {'method': file.read(methodsize)}
        else:
                c = {'method': b'\x00'}
```

**Comment by xVanTuring** [comment]
@miurahr It works now! Thanks!
